src/main/resources/buildRunner.m

```diff
@@ -0,0 +1,37 @@
+
+% This is the MATLAB script which generates the JSON file.
+function  buildrunner(varargin)
+p= buildfile;
+if isempty(varargin)
+    res = p.run();
+else
+    s = string(varargin);
+    a = split(s);
+    tasks = [];
+    for idx = 1:length(a)
+        tasks = [tasks,a(idx)];
+    end
+    res= p.run(tasks);
+end
+
+fID = makeFile();
+taskDetails = struct();
+for idx = 1:numel(res.TaskResults)
+    taskDetails(idx).name = res.TaskResults(idx).Name;
+    taskDetails(idx).description = p(res.TaskResults(idx).Name).Description;
+    taskDetails(idx).failed = res.TaskResults(idx).Failed;
+    taskDetails(idx).skipped = res.TaskResults(idx).Skipped;
+    taskDetails(idx).duration = string(res.TaskResults(idx).Duration);
+end
+a = struct("taskDetails",taskDetails);
+s = jsonencode(a,"PrettyPrint",true);
```

Comment:
Seems like we wouldn't need to pretty print a file only intended for being read by the machine?

Reply:
Yeah we do not need as such but it would help one read it whenever needed , as we are still keeping it undr build/root directory. I can remove it if t has any known performance implications.